test/adapters/CharmAdapter.js

```diff
@@ -0,0 +1,930 @@
+const { assert, expect } = require("chai");
+const { ethers } = require("hardhat");
+const { provider, BigNumber, constants } = ethers;
+const { parseEther } = ethers.utils;
+
+const time = require("../helpers/time.js");
+const {
+  wmul,
+  wdiv,
+} = require("../helpers/utils");
+
+const CHARM_OPTION_FACTORY = "";
+const CHARM_OPTION_VIEW = "";
+
+const WAD = BigNumber.from("10").pow(BigNumber.from("18"));
+
+const WETH_ADDRESS = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+const USDC_ADDRESS = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+const WBTC_ADDRESS = "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599";
+const ETH_ADDRESS = constants.AddressZero;
+const ONE_ADDRESS = "0x0000000000000000000000000000000000000001";
+let owner, user, recipient;
+let ownerSigner;
+
+const PUT_OPTION_TYPE = 1;
+const CALL_OPTION_TYPE = 2;
+
+
+// NOTE: USE WITH BLOCKNUM 12032283 when FORKING
+
+describe.skip("CharmAdapter", () => {
```

Comment:
remove .skip when testing locally

Reply:
I just learned that we can use `hardhat_reset`. https://hardhat.org/guides/mainnet-forking.html#resetting-the-fork

Can you try it out?